haskell_scotty/FetchClients.hs

```diff
@@ -0,0 +1,20 @@
+{-# LANGUAGE OverloadedStrings #-}
+module FetchClients where
+
+import Client
+import DBConfig
+import Database.MySQL.Simple
+import Control.Monad
+
+-- | Fetch a client from MySQL.
+fetchClients :: IO [Client]
+fetchClients = do
+  DBConfig user pass database <- readDBConfig
+  conn <- connect defaultConnectInfo {
+            connectUser = user,
+            connectPassword = pass,
+            connectDatabase = database }
+  results <- query_ conn "SELECT * FROM clients"
+  clients <- forM results $ \(id, name, address, city, state, zip, country, email, contact, phone, created_at, updated_at) ->
+    return $ Client id name address city state zip country email contact phone created_at updated_at
```

Comment:
`query_` returns a `QueryResults` kind of value. `Database.MySQL.Simple` only provides automatic conversions from this into a tuple, which as you have observed is error-prone and not terribly useful. One could presumably define something more sophisticated, but it would take some work. The right plan here is probably to make `Client` be an instance of `QueryResults` and define `convertResults` to do appropriate checking and whatnot.